src/azul/terraform.py

```diff
@@ -776,6 +781,34 @@ def tf_config(self, app_name):
             resource['source_code_hash'] = '${filebase64sha256("%s")}' % package_zip
             resource['filename'] = package_zip
 
+        # Replace any references to unqualified function ARNs emitted by Chalice
+        # with references to the alias. This must be happen before the injection
+        # of the alias resources, else the aws_lambda_alias.function_name fields
+        # would be affected by the regex substitution.
```

Comment:
```suggestion
        # Replace any references to unqualified function ARNs emitted by Chalice
        # with references to the alias. This must be happen before the injection
        # of the alias resources, or else the aws_lambda_alias.function_name fields
        # would be affected by the regex substitution.
```

However, please try moving the alias resources to the explicit configuration, same place as `aws_lambda_function_event_invoke_config`. Then this constraint goes away.

Reply:
Moving the `aws_lambda_alias` resources from `tf_config()` to `api_gateway.tf.json.template.py` would require adding a function that duplicates work done by `tf_config()` to read the indexer/service `chalice.tf.json`, since we need all the the names of `aws_lambda_function` for the aliases.

Reply:
I was able to remove the constraint however by populating `aws_lambda_alias.function_name` with a reference to the `.function_name` property of `aws_lambda_function` instead of `.arn` .

Reply:
> Moving the aws_lambda_alias resources from … since we need all the the names of aws_lambda_function for the aliases

See if we can use the functionality we implemented for retries to list handler names.